client/src/main/java/com/netflix/conductor/client/automator/TaskPollExecutor.java

```diff
@@ -306,13 +304,22 @@ private void updateTaskResult(int count, Task task, TaskResult result, Worker wo
     }
 
     private TaskResult retryOperation(
-            Function<TaskResult, Void> operation, int count, TaskResult result) {
+            Function<TaskResult, Optional<String>> operation, int count, TaskResult result) {
         int index = 0;
         while (index < count) {
             try {
-                TaskResult taskResult = result.copy();
-                operation.apply(taskResult);
-                return taskResult;
+                Optional<String> optionalExternalStorageLocation = operation.apply(result);
+                if (optionalExternalStorageLocation.isPresent()) {
```

Comment:
This is confusing. Can we create two retryOperation method? one for the uploadtoexternalstorage, which takes a taskresult output map and returns an externalstoragepath, another one for the updatetask, which takes the final taskresult from last step, and return void

Reply:
If we want to keep just one `retryoperation` method, we can just remove line #313 `TaskResult taskResult = result.copy();` from it. I don't see the point of why we want to make a copy of it for every retry